### ISSUE TITLE
Start providing Windows releases

### ISSUE THREAD

**Issue by devonbleak** [segmentio/aws-okta#119]
With the issues Go has on Windows with network drives and the recent addition of `gcc` as a build dependency and `libconv.dll` as a runtime dependency it's getting to the point that we can't reasonably expect users to all be building on Windows.

I'd be willing to start building/providing 64-bit WIndows binaries in a .zip or something if we can figure out how to incorporate it into the GitHub releases.

Ideally I think we'd get something like a .msi or chocolatey package built but that's probably beyond what I can commit to personally for now.

**Comment by mlehner616** [comment]
+1

**Comment by RafPe** [comment]
I would suggest then something really simple as https://github.com/goreleaser/goreleaser in conjunction with Travis CI. 

Have tested that and works flawlessly generating binaries for Linux/Windows and MacOS ( of course depending on config setup )

**Comment by nickatsegment** [comment]
> I would suggest then something really simple as [goreleaser/goreleaser](https://github.com/goreleaser/goreleaser) in conjunction with Travis CI.

Currently we use CircleCI for CI, plain old go to do binary builds, and github-release to do the upload. I don't see an advantage to switching to goreleaser, though I haven't used it before. 

We'd have to add a new windows target in the Makefile. Because we rely on some C libs (USB HID) so I'm pretty sure we'd need to actually build on Windows, which [CircleCI only just added early-access support for on June 21](https://discuss.circleci.com/t/windows-early-access-now-available-on-circleci/30977), unless we figured out a cross-compile toolchain.
 	
Basically, this is all a bit of work and figuring out for a platform that Segment doesn't really use, and would be hard to offer any support or testing around. Hence the `help wanted` label.

**Comment by dkujawski** [comment]
Hello, 

Are you still using CircleCI to manage your build processes?

I have been working on getting a CircleCI build setup to generate a Windows executable and MSI.  I have been able to successfully build the raw aws-okta executable that runs without the need to use the Win-builds process to install dependent libraries.  

I am currently working in a fork but would like to put a PR together once I get things cleaned up and the packaging, MSI, component in order.  

Aside from what is outlined in the contributing guidelines, is there anything that I need to be aware of or discussions that should be had?

-Dave

**Comment by eculver** [comment]
Hi @dkujawski, thanks for helping out. I'll quote @nickatsegment from above
> We'd have to add a new windows target in the Makefile. Because we rely on some C libs (USB HID) so I'm pretty sure we'd need to actually build on Windows, which CircleCI only just added early-access support for on June 21, unless we figured out a cross-compile toolchain.

So, if you are familiar with Windows builds on CircleCI and are willing to cook up some config to get us rolling that would be splendid. I'm happy to review once it's ready! 

Thanks!
